### ISSUE TITLE
Child selectors not applied correctly in selectNodesWithEq

### ISSUE THREAD

**Issue by paul-bjorkstrand** [adobe/alloy#850]
When `selectNodesWithEq` emulates `eq()`, the child selectors adjacent to the `eq()`s are stripped off, causing incorrect selection.

### Given
HTML:
```html
<body>
  <div id="target">
    <div>
      <div>
        <span id="wrong"></span>
      </div>
      <span id="right"></span>
    </div>
  </div>
</body>
```

Selector:
```
#target > div:eq(0) > span
```

### Expected Behaviour
When `selectNodesWithEq` is called with the above selector and markup in the document, it should select `<span id="right"></span>` as it is the only element that matches `#target > div > span`.

### Actual Behaviour
When `selectNodesWithEq` is called with the above selector and markup in the document, it actually selects `<span id="wrong"></span>`

#### Platform and Version
Tested in Chrome (version 98+), but presumed to be an issue in all browsers

alloy.js info
```json
{
  "libraryInfo": {
    "version": "2.9.0"
  }
}
```

#### Sample Code that illustrates the problem
Add the following test case to `selectNodesWithEq.spec.js`
```javascript
  it("should respect child selectors", () => {
    const content = `
      <div>
        <div>
            <span id="wrong"></span>
        </div>
        <span id="right"></span>
      </div>
    `;

    const node = createNode(
      "DIV",
      { id: "abc", class: "eq" },
      { innerHTML: content }
    );

    appendNode(document.body, node);

    const result = selectNodesWithEq("#abc > div:eq(0) > span");

    expect(result[0].tagName).toEqual("SPAN");
    expect(result[0].id).toEqual("right");
  });
```

**Comment by ninaceban** [comment]
Thank you for finding the issue and creating the test case. Apparently, we have had this bug from the beginning. 

For some background, the node selection functionality has been partially ported from Target At.js.  At.js uses [Zepto](https://www.npmjs.com/package/zepto) library for DOM query. 
It is interesting to note that `Zepto` uses a hack when it encounters a selector like ` > span` it will change it to something like `.Zepto12345 > span` and this ensures that the right children are selected - [please check Zepto source code](https://github.com/madrobby/zepto/blob/main/src/selector.js#L62).
To keep the implementation as close as possible to At.js we will follow the same approach. This should allow us to maintain cross browser support.

@jfkhoury @jonsnyder

**Comment by carterworks** [comment]
Fixed with #858
